### PR TITLE
IBX-8858: Update Solr config between 4.5 and 4.6

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-8858](https://issues.ibexa.co/browse/IBX-8858)
| Versions      | 4.6
| Edition       | All

Update Solr config when upgrading from 4.5 to 4.6 to enable spellchecking.

Preview: https://ez-systems-developer-documentation--2486.com.readthedocs.build/en/2486/update_and_migration/from_4.5/update_from_4.5/#update-solr-configuration

ibexa/solr#52 was merged into v4.6.0-beta4

```diff
diff -r v4.5.7/solrconfig.xml v4.6.10/solrconfig.xml
767a768,770
>     <arr name="last-components">
>       <str>spellcheck</str>
>     </arr>
911c914
<       <str name="field">_text_</str>
---
>       <str name="field">meta_content__text_t</str>
```

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 1]
```diff
@@ -413,6 +413,12 @@ The following migration example allows users with the `Editor` role to access th
                     values: []
 ```
 
+## Update Solr configuration
+
+Solr configuration changes with the addition of new features such as spellchecking.
+
+[Generate the new configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration).
```
Comment (adamwojs) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:420]: Solr configuration might be customized by 3rd-party, so we need to describe actual changes. As developer I would expect at least the same level of detail as provided in PR description: https://github.com/ibexa/solr/pull/52.

**Review by adamwojs — CHANGES_REQUESTED** [review]
We need to be more

File: docs/update_and_migration/from_4.5/update_from_4.5.md (hunk 1 above)
Comment (adriendupuis) [reply docs/update_and_migration/from_4.5/update_from_4.5.md]: e93fb0cf19581dbc9c84852aa86eccd078f9ef4d: I now expose the XML of the component and its assignation to /select; I moved the `generate-solr-config.sh` trick into a note box.

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 2]
```diff
@@ -413,6 +413,45 @@ The following migration example allows users with the `Editor` role to access th
...
+      <int name="maxEdits">2</int>
+      <int name="minPrefix">1</int>
+      <int name="maxInspections">5</int>
+      <int name="minQueryLength">4</int>
+      <float name="maxQueryFrequency">0.01</float>
+    </lst>
+  </searchComponent>
+```
+
+Add this `spellcheck` component to `/select`:
```
Comment (vidarl) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:439]:
```suggestion
Add this `spellcheck` component to the `/select` request handler: 
```

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 3]
```diff
@@ -413,6 +413,45 @@ The following migration example allows users with the `Editor` role to access th
                     values: []
 ```
 
+## Update Solr configuration
+
+Solr configuration changes with the addition spellchecking feature.
+
+Configure the `spellcheck` component:
```
Comment (vidarl) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:420]:
```suggestion
Configure the `spellcheck` component in `solrconfig.xml`:
```

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 4]
```diff
@@ -413,6 +413,45 @@ The following migration example allows users with the `Editor` role to access th
...
+    </arr>
+    <!-- […] -->
+  </requestHandler>
+```
+
+!!! note
+
+    You can [generate new Solr configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration),
+    and compare with your current configuration to merge those `spellcheck` configurations.
+
```
Comment (vidarl) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:454]:
```suggestion

Once `solrconfig.xml` has been changed, solr needs to be restarted for the changes to take affect

```

**Review by vidarl — CHANGES_REQUESTED** [review]

File: docs/update_and_migration/from_4.5/update_from_4.5.md (hunk 4 above)
Comment (adriendupuis) [reply docs/update_and_migration/from_4.5/update_from_4.5.md:454]:
```suggestion

Restart Solr for `solrconfig.xml` changes to take effect.

```

**Review by vidarl — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 5]
```diff
@@ -413,6 +413,47 @@ The following migration example allows users with the `Editor` role to access th
                     values: []
 ```
 
+## Update Solr configuration
+
+Solr configuration changes with the addition spellchecking feature.
```
Comment (julitafalcondusza) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:418]:
```suggestion
Solr configuration changes with the addition of spellchecking feature.
```

File: docs/update_and_migration/from_4.5/update_from_4.5.md [hunk 6]
```diff
@@ -413,6 +413,47 @@ The following migration example allows users with the `Editor` role to access th
...
+      <str>spellcheck</str>
+    </arr>
+    <!-- […] -->
+  </requestHandler>
+```
+
+!!! note
+
+    You can [generate new Solr configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration),
+    and compare with your current configuration to merge those `spellcheck` configurations.
```
Comment (julitafalcondusza) [inline docs/update_and_migration/from_4.5/update_from_4.5.md:453]:
```suggestion
    and merge `spellcheck` configuration by comparing new files with your existing setup.
```

**Review by julitafalcondusza — CHANGES_REQUESTED** [review]
Please check my comments.

**Review by julitafalcondusza — APPROVED** [review]

**Pull request merged**